tests/README.md

```diff
@@ -8,91 +8,223 @@ Priority is to provide very simple system of test scenarios, easily extendable
 so everyone contributing remediation can also provide cases where this
 remediation works.
 
-## Prerequisites
+The Test Suite executes the scans, remediations, and test scenarios on VMs or containers.\
+To test content using VMs, *Libvirt backend* is used;\
+to test content using containers, either *Podman* or *Docker* backend can be used.
+
+Once a domain or container is prepared it can be used indefinitely.\
+The Test Suite can perform tests focused on a profile or a specific rule.\
+Typically, for a test run, a sequence of scan, remediation and scan is peformed.
+
+## How to prepare a backend for testing
+For the Test Suite to work, you need to have a backend image prepared for testing.\
+You can use a powerful full-blown VM backend, or a lightweight container backend.
+
+SSG Test Suite currently does not provide automated provisioning of VMs or containers.
+
+### Libvirt backend
+
+To use Libvirt backend, you need to have:
+
+- These packages installed in your host
+  - `openssh-clients`
+  - `libvirt`
+  - `libvirt-daemon`
+  - `python2-libvirt/python3-libvirt`
+  - `virt-install`      (recommended, used by `install_vm.py` script)
+  - `libvirt-client`    (optional, to manage VMs via console)
+  - `virt-manager`      (optional, to manage VMs via GUI)
+  - `virt-viewer`       (optional, to access graphical console of VMs)
+  - `ansible`           (required if remediating via Ansible)
+- A VM that fullfils following requirements:
+  - Package `qemu-guest-agent` installed
+  - Package `openscap` version 1.2.15 or higher installed
+  - `root` can login via ssh (it is recommended to setup key-based authentication)
+  - `root` can install packages (for RHEL7, it means subscription enabled).
+
+An easy way to install your VM is via `install_vm.py` script. It will setup a VM\
+according to the requirements, including configuration of a key for SSH login.
+
+Common usage looks like:
+```
+$ ./install_vm.py --domain test-suite-fedora --distro fedora
+```
+
+By default, the key at `~/.ssh/id_rsa.pub` is used. You can change default key used via `--ssh-pubkey` option.
+
+By default, the VM will be created on the user hypervisor, i.e. `qemu:///session`.
+For TestSuite this should be enough, in case you need the VM to reside under `qemu:///system`, use the install script with `--libvirt qemu:///system`.
+
+When installing a RHEL VM, you will still need to subscribe it.
+
+*TIP*: Create a snapshot as soon as your VM is setup. This way, you can manually revert
```

Comment:
Maybe we can mention here that snapshots can be created using `virsh(1)` to point a reader where to look.

Reply:
Added a note.